schema/crdb/populate-db-metadata-nexus/up04.sql

```diff
@@ -0,0 +1,15 @@
+-- Populate db_metadata_nexus records for all Nexus zones in the current target blueprint
+--
+-- This migration handles backfill for existing deployments that are upgrading
+-- to include db_metadata_nexus. It finds all Nexus zones in the current
+-- target blueprint and marks them as 'active' in the db_metadata_nexus table.
+
+SET LOCAL disallow_full_table_scans = off;
+
+INSERT INTO omicron.public.db_metadata_nexus (nexus_id, last_drained_blueprint_id, state)
```

Comment:
It may be possible to not include this populate step, depending on how reconfigurator execution plans on populating these records. After all, the "no `db_metadata_nexus` records" case is already treated specially for backwards-compatibility.

This woula slo let us delete the data migrations in `nexus/tests/integration_tests/schema.rs`